### DISCUSSION TITLE
Can rdkit convert svg file into smile format?

### DISCUSSION THREAD

**Discussion by DooYal** [rdkit/rdkit#5936] [Q&A]
Like this compound in http://www.tcmip.cn/ETCM/index.php/Home/Index/cf_details.html?id=2283, where there is no Pubchemid or CASid, and there is only a 2D structure image with some values from chemical calculation. When I opened the source file, I found this 2D structure is a svg file whose address is http://www.tcmip.cn/ETCM/Public/SDF/molecular/TCMIP102342.svg. So, is it available to convert the svg file into smile format?

**Comment by DavidACosgrove** [comment]
Hi,
There is nothing in RDKit to do this, AFAIK.  It is an interesting but
poorly defined problem.  There is probably enough information in the SVG
you have provided to build the structure by analysing where all the lines,
text, etc are.  This would not be a general solution for all SVGs though,
because there are many different ways of drawing lines, text etc.
ChemDraw, for example, draws all lines, even single bonds, as polygons
rather than simple lines, and RDKit SVGs have, by default, the characters
drawn as freetype glyphs so there are no obvious chemical symbols in the
SVG file itself.
There are several open source systems available for converting images of
chemical structures into a 2D connection table.  Try googling "convert
image to chemical structure".  To some extent this is a backward approach
since you would have to take a screenshot of the image to pass into such
software, where the first thing it will try and do is work out where all
the bonds are from the black and white pixels...

Dave


On Thu, Jan 5, 2023 at 11:10 AM DooYal ***@***.***> wrote:

> Like this compound in
> http://www.tcmip.cn/ETCM/index.php/Home/Index/cf_details.html?id=2283,
> where there is no Pubchemid or CASid, and there is only a 2D structure
> image with some values from chemical calculation. When I opened the source
> file, I found this 2D structure is a svg file whose address is
> http://www.tcmip.cn/ETCM/Public/SDF/molecular/TCMIP102342.svg. So, is it
> available to convert the svg file into smile format?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/5936>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FV56OKRQRSS5J5P7JDWQ2T4DANCNFSM6AAAAAATR2OGRE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Reply by DooYal** [reply]
Thanks for the suggestion. I will try to google "convert image to chemical structure".
